Source/WebCore/html/CanvasBase.cpp

```diff
@@ -380,12 +382,12 @@ RefPtr<ImageBuffer> CanvasBase::allocateImageBuffer(bool usesDisplayListDrawing,
 bool CanvasBase::shouldInjectNoiseBeforeReadback() const
 {
     // Note, every early-return resulting from this check potentially leaks this state. This is a risk that we're accepting right now.
-    return m_imageBuffer && scriptExecutionContext() && scriptExecutionContext()->noiseInjectionHashSalt();
+    return !!m_canvasNoiseHashSalt;
 }
 
 bool CanvasBase::postProcessPixelBufferResults(Ref<PixelBuffer>&& pixelBuffer) const
 {
-    if (auto noiseInjectionHashSalt = shouldInjectNoiseBeforeReadback() ? scriptExecutionContext()->noiseInjectionHashSalt() : std::nullopt)
+    if (auto noiseInjectionHashSalt = m_canvasNoiseHashSalt)
         return m_canvasNoiseInjection.postProcessPixelBufferResults(std::forward<Ref<PixelBuffer>>(pixelBuffer), *noiseInjectionHashSalt);
```

Comment:
Super minor nit - I'm not sure the separate local variable adds much clarity.

```suggestion
    if (m_canvasNoiseHashSalt)
        return m_canvasNoiseInjection.postProcessPixelBufferResults(std::forward<Ref<PixelBuffer>>(pixelBuffer), *m_canvasNoiseHashSalt);
```